library/src/network.rs

```diff
@@ -56,20 +56,39 @@ impl Default for NetworkHooks {
     }
 }
 
+fn create_client_with_proxy() -> anyhow::Result<reqwest::blocking::Client> {
+    let config = read_config()?;
+    let mut client_builder = reqwest::blocking::Client::builder();
+    
+    if let Some(proxy_url) = config.proxy_url {
+        let proxy = reqwest::Proxy::http(&proxy_url)?;
+        client_builder = client_builder.proxy(proxy);
+    }
+    
+    let client = client_builder.build()?;
+    Ok(client)
+}
+
+fn read_config() -> anyhow::Result<YamlConfig> {
```

Comment:
I think typically config is set through https://github.com/shorebirdtech/updater/blob/main/library/src/config.rs#L91

Reply:
I see, looks like the request callbacks would now need to be config-specifc. 🤔